### PR TITLE
[Cleanup] storage/s3-sdk: Adds docstrings, return value instead of print 

### PR DESCRIPTION
**Pull request by askmeegs**
## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: storage/s3-sdk/list_gcs_buckets.py [hunk 1]
```diff
@@ -21,11 +21,19 @@
...
+    google_access_key_secret      -- HMAC access secret
+
+    Returned value is a list of strings, one for each bucket name.
+
+    To use this sample:
+    1. Create a Cloud Storage HMAC key: https://cloud.google.com/storage/docs/authentication/managing-hmackeys#create
+    2. Change endpoint_url to a Google Cloud Storage XML API endpoint.
+
+    To learn more about HMAC: https://cloud.google.com/storage/docs/authentication/hmackeys#overview
+    """
```
Comment (dandhlee) [inline storage/s3-sdk/list_gcs_buckets.py:36]: Please follow the docstring format as shown in https://google.github.io/styleguide/pyguide#383-functions-and-methods.
```suggestion
    """Lists all Cloud Storage buckets using AWS SDK for Python (boto3)

    Positional arguments:
    google_access_key_id          -- hash-based message authentication code (HMAC) access ID
    google_access_key_secret      -- HMAC access secret

    To use this sample:
    1. Create a Cloud Storage HMAC key: https://cloud.google.com/storage/docs/authentication/managing-hmackeys#create
    2. Change endpoint_url to a Google Cloud Storage XML API endpoint.

    To learn more about HMAC: https://cloud.google.com/storage/docs/authentication/hmackeys#overview
    
    Returns:
        A list of strings, one for each bucket name.
    """
```

File: storage/s3-sdk/list_gcs_objects.py [hunk 2]
```diff
@@ -21,12 +21,20 @@
 
 
 def list_gcs_objects(google_access_key_id, google_access_key_secret, bucket_name):
-    """Lists GCS objects using boto3 SDK"""
```
Comment (dandhlee) [inline storage/s3-sdk/list_gcs_objects.py:24]: Ditto.

File: storage/s3-sdk/list_gcs_buckets.py [hunk 3]
```diff
@@ -21,11 +21,19 @@
 
 
 def list_gcs_buckets(google_access_key_id, google_access_key_secret):
```
Comment (dandhlee) [inline storage/s3-sdk/list_gcs_buckets.py:23]: Don't forget docstrings for the parameters.

**Review by dandhlee — CHANGES_REQUESTED** [review]
Are there any reason to switch to returning a list rather than printing them?

**Comment by engelke** [comment]
> Are there any reason to switch to returning a list rather than printing them?

Yes. We want to test against returned values rather than captured strings.

File: storage/s3-sdk/list_gcs_buckets.py (hunk 3 above)
Comment (engelke) [reply storage/s3-sdk/list_gcs_buckets.py:23]: See PR #9947

File: storage/s3-sdk/s3_sdk_test.py [hunk 4]
```diff
@@ -69,24 +69,20 @@ def test_blob(test_bucket):
 # to 15s.
 @backoff.on_exception(backoff.constant, ClientError, interval=1, max_time=15)
 def test_list_buckets(capsys, hmac_fixture, test_bucket):
```
Comment (engelke) [inline storage/s3-sdk/s3_sdk_test.py:71]: Can remove capsys now.

File: storage/s3-sdk/s3_sdk_test.py [hunk 5]
```diff
@@ -69,24 +69,20 @@ def test_blob(test_bucket):
...
-    out, _ = capsys.readouterr()
-    assert "Buckets:" in out
-    assert test_bucket.name in out
+    assert test_bucket.name in results
 
 
 # Retry request because the created key may not be fully propagated for up
 # to 15s.
 @backoff.on_exception(backoff.constant, ClientError, interval=1, max_time=15)
 def test_list_blobs(capsys, hmac_fixture, test_bucket, test_blob):
```
Comment (engelke) [inline storage/s3-sdk/s3_sdk_test.py:82]: Can remove capsys

**Review by engelke — APPROVED** [review]
Please remove "capsys" parameters and fix lint's nitpicking. You don't need to wait for another review from me. LGTM.

**Comment by tritone** [comment]
> > Are there any reason to switch to returning a list rather than printing them?
> 
> Yes. We want to test against returned values rather than captured strings.

Have we changed that in general across samples? I see this in the standards still: https://googlecloudplatform.github.io/samples-style-guide/#result

**Comment by engelke** [comment]
> > > Are there any reason to switch to returning a list rather than printing them?
> > 
> > 
> > Yes. We want to test against returned values rather than captured strings.
> 
> Have we changed that in general across samples? I see this in the standards still: https://googlecloudplatform.github.io/samples-style-guide/#result

The instructions for FixIt don't require removing printing, just no longer basing tests on that instead of on returned values.

**Review by dandhlee — APPROVED** [review]

**Pull request merged**